programs/x509/cert_write_demo.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/sh
+set -e -u
+
+sed 's/^/# /' <<'EOF'
+This script demonstrates the use of x509 tools to generate a certificate
+for a private key. It shows both the case of a self-signed certificate and
+the case of a certificate signed by a different key using a certificate
+signing request.
+EOF
+
+programs="${0%/*}"; programs="${programs%/*}"
+
+ca_key="demo_ca.key"
+server_key="demo_server.key"
+csr="demo_csr.req"
+ca_crt="demo_ca.crt"
+server_crt="demo_server.crt"
+
+run () {
+    echo
+    echo "# $1"
+    shift
+    echo "+ $*"
+    "$@"
+}
+
+run_openssl () {
```

Comment:
Replying to the commit message: "Use OpenSSL if available to show the certificate contents, because we
don't have a tool to do that." - how about `programs/x509/cert_app mode=file filename=<name>`? Granted, the output's not as neat or as complete, but it's something, it avoids a dependency, and at as a bonus it smoke-tests one more program.

Reply:
If you think our tool's output really is not enough, perhaps use our tool first, and then optionally OpenSSL in addition when available?

Reply:
I must have missed `cert_app` when I wrote that. I don't think the things that it doesn't show are useful here.